src/optim/FishLeg/fishleg.py

```diff
@@ -395,15 +362,21 @@ def update_aux(self) -> None:
         self.aux_opt.step()
         return aux_loss.item()
 
-    # TODO: What do we need here? - This method is taken from Adam, to discuss!
+    def _augment_params_by(self, eps: float, nat_grads: List):
+        p_idx = 0
+        for module in self.model.modules():
+            if isinstance(module, FishModule):
+                for name, param in module.named_parameters():
+                    if "fishleg_aux" not in name:
+                        param.data -= 2 * eps * nat_grads[p_idx]
```

Comment:
This function looks good but I think it might be worth defining it as small function snipped inside the update_aux fnction (instead of defining it as a self function); if you did that you would not need to pass the nat_grads either as input and also we are not going to use it anywhere else anyway. Anyway it is just a small detail, here it is fine too. However, according to how you used this function above you should not be using the form 
```
param.data -= 2 * eps * nat_grads[p_idx]
```
but you should be using the form 
```
param.data +=  eps * nat_grads[p_idx]
```
otherwise I think you are introducing a bug, from what I can see